Remove snappy from base Ubuntu system as part of installation process?

## Description

Ubuntu 16.04 includes the snappy package manager, an alternative package management system that allows applications to be installed along with their dependencies in a single bundle. It's not used by SecureDrop and could be removed during the initial server setup, thereby reducing the overall attack surface.

## User Research Evidence

None, this is not a user-facing issue.

## User Stories

As a SecureDrop admin, I want the system to only include tools and packages required by SecureDrop.

Seeing a test failure on Focal for this, but only on Qubes staging VMs, created from ISO. Test failure looks like:

```
host = <testinfra.host.Host ansible://app-staging>
    
        def test_aa_no_denies_in_syslog(host):
            """ Ensure that there are no apparmor denials in syslog """
            with host.sudo():
                f = host.file("/var/log/syslog")
    >           assert 'apparmor="DENIED"' not in f.content_string
    E           assert 'apparmor="DENIED"' not in 'Feb 11 14:2...uildtimes.\n'
    E             'apparmor="DENIED"' is contained here:
    E               Feb 11 14:29:19 app-staging rsyslogd: [origin software="rsyslogd" swVersion="8.2001.0" x-pid="640" x-info="https://www.rsyslog.com"] rsyslogd was HUPed
    E               Feb 11 14:29:19 app-staging dbus-daemon[628]: dbus[628]: Unknown group "power" in message bus configuration file
    E               Feb 11 14:29:19 app-staging systemd[1]: Starting Hold until boot process finishes up...
    E               Feb 11 14:29:19 app-staging systemd[1]: Starting Terminate Plymouth Boot Screen...
    E               Feb 11 14:29:19 app-staging snapd[642]: daemon.go:347: started snapd/2.48+20.04 (series 16; classic) ubun...
    E             
    E             ...Full output truncated (4770 lines hidden), use '-vv' to show
    
    ../testinfra/app/test_apparmor.py:128: AssertionError


```

The specific lines causing the test to fail are:

```
sdadmin@app-staging:~$ sudo grep -F DENIED /var/log/syslog
Feb 11 14:34:13 app-staging kernel: [   63.945397] audit: type=1400 audit(1613054053.529:46): apparmor="DENIED" operation="open" profile="/usr/sbin/ntpd" name="/snap/bin/" pid=2239 comm="ntpd" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Feb 11 14:36:53 app-staging kernel: [  224.349950] audit: type=1400 audit(1613054213.933:47): apparmor="DENIED" operation="open" profile="/usr/sbin/ntpd" name="/snap/bin/" pid=22457 comm="ntpd" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
```

We have a PR up in #5784, only documenting this to point out that it's another small divergence between ISOs and Vagrant boxes.

--- a/install_files/ansible-base/roles/common/defaults/main.yml
+++ b/install_files/ansible-base/roles/common/defaults/main.yml
@@ -49,3 +49,4 @@ unused_packages:
   - libiw30
   - wireless-tools
   - wpasupplicant
+  - snapd

--- a/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
+++ b/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
@@ -3,6 +3,7 @@
   apt:
     name: "{{ unused_packages }}"
     state: absent
+    purge: yes
   tags:
     - apt
     - hardening

--- a/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
+++ b/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
@@ -7,6 +7,12 @@
     - apt
     - hardening
 
+# Remove the apparmor config file for snapd
+- name: Remove snapd config for apparmor
+  file:
+    path: /etc/apparmor.d/usr.lib.snapd.snap-confine.real
+    state: absent
+
 # After installing securedrop-grsec, remove
 # old generic kernels to avoid accidental
 # boots into a less secure environment.

--- a/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
+++ b/install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml
@@ -7,12 +7,6 @@
     - apt
     - hardening
 
-# Remove the apparmor config file for snapd
-- name: Remove snapd config for apparmor
-  file:
-    path: /etc/apparmor.d/usr.lib.snapd.snap-confine.real
-    state: absent
-
 # After installing securedrop-grsec, remove
 # old generic kernels to avoid accidental
 # boots into a less secure environment.

--- a/molecule/testinfra/common/test_system_hardening.py
+++ b/molecule/testinfra/common/test_system_hardening.py
@@ -148,8 +148,9 @@ def test_no_ecrypt_messages_in_logs(host, logfile):
 @pytest.mark.parametrize('package', [
     'cloud-init',
     'libiw30',
-    'wpasupplicant',
+    'snapd',
     'wireless-tools',
+    'wpasupplicant',
 ])
 def test_unused_packages_are_removed(host, package):
     """ Check if unused package is present """
@@ -165,3 +166,10 @@ def test_iptables_packages(host):
         assert host.package("iptables-persistent").is_installed
     else:
         assert not host.package("iptables-persistent").is_installed
+
+
+def test_snapd_absent(host):
+    assert not host.file("/lib/systemd/system/snapd.service").exists
+    assert not host.file("/etc/apparmor.d/usr.lib.snapd.snap-confine.real").exists
+    assert not host.file("/usr/bin/snap").exists
+    assert not host.file("/var/lib/snapd/snaps").exists

